Sources/CoreMetrics/Metrics.swift

```diff
@@ -12,12 +12,15 @@
 //
 //===----------------------------------------------------------------------===//
 
+public protocol MetricHandler: AnyObject {
```

Comment:
This unlocks being able to do an API like: `release(metric: MetricHandler)`
Implementations need to `switch` anyway internally since they most likely want to get their internal types ("TestCounterHandler" etc) to get the identity that should be released